org.osgi.test.cases.cm/src/org/osgi/test/cases/cm/junit/CMCoordinationTestCase.java

```diff
@@ -436,8 +436,11 @@ public void updated(Dictionary<String, ? > properties)
 
 		// wait and verify listener
 		sleep();
-		assertEquals(1, events.size());
-		assertFalse(events.get(0));
+		assertTrue(events.size() <= 2);
+		// all events must be 'null' updates
+		for (Boolean b : events) {
```

Comment:
would it make sense to to additionally add an addition `assertEquals(1, events.size());` right after `registerService(..)` but outside coordination, to make it clearer that updated() should have been called?

<img width="818" alt="image" src="https://github.com/osgi/osgi/assets/188422/1ca2d206-3bfe-435f-bbc9-93f4bd27a8a5">

Reply:
I have adjusted the testcase to be more expressive now.